product-service/src/main/java/com/minhaz/orders101/endpoints/ProductsEndpoint.java

```diff
@@ -103,6 +107,46 @@ private Object notFoundError(String id) {
     })).build();
   }
 
+  @PATCH
+  @Path("stock-availability/{productId}")
+  @Consumes({"application/json"})
+  @Produces({"application/json"})
+  public Response updateStock(@PathParam("productId") String productId, @QueryParam("inc") boolean inc,
+      @QueryParam("qty") Integer qty) {
+    Optional<Product> product = productService.retrieveById(productId);
+    if (qty == null) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(quantityNegativeNullError(null)).build();
+    }
+    if (product.isEmpty()) {
+      return Response.status(Response.Status.NOT_FOUND).entity(notFoundError(productId)).build();
+
+    }
+    if (product.isPresent() & qty > 0) {
+      int stockLevel = product.get().getStockLevel();
+      StockAvailability stockAvailability = StockAvailability.builder().id(productId).requestQuantity(qty)
+          .isAvailable(checkAvailability(stockLevel, qty)).build();
+      if (!inc && stockAvailability.isAvailable()) {
+        product.get().setStockLevel(stockLevel - qty);
+        productService.persist(product.get());
+      } else if (inc) {
+        product.get().setStockLevel(stockLevel + qty);
+        productService.persist(product.get());
+      }
+      StockAvailability stockAvailabilityLatest = StockAvailability.builder().id(productId).requestQuantity(qty)
+          .isAvailable(checkAvailability(stockLevel, qty)).build();
+      return Response.ok().entity(ResponseModel.builder().data(stockAvailabilityLatest).build()).build();
+    }
+    return Response.status(Response.Status.BAD_REQUEST).entity(quantityNegativeNullError(qty)).build();
+  }
+
+  private Object quantityNegativeNullError(Integer qty) {
+    return ResponseModel.builder().errors(List.of(new HashMap<>() {
+      {
+        put("error", String.format(QTY_CAN_NOT_BE_A_NEGATIVE_NULL_VALUE, qty));
```

Comment:
if quantity is null, is there much value in adding to the output message? 🤔 

Ignoring the repeated checking of the qty param variable, can you write something here that's a bit more elegant and makes better use of the framework/ libraries you have been learning and applying in the orders service? Hint: you don't need to test the qty value _inside_ method - you can still feedback to the client if their input is not acceptable.